pkg/satellite/stream/stream.go

```diff
@@ -178,6 +191,18 @@ func (ss *satelliteStream) recvLoop() {
 	b := backoff.NewExponentialBackOff()
 	b.MaxElapsedTime = MaxElapsedTime
 
+	// Initialize auto close timer
+	receivingBytes := false
+	var timestampLastByteReceived *timestamp.Timestamp
+	autoCloseTimer := time.AfterFunc(ss.autoCloseDelay, func() {
+		// Timer reached the end of the delay period. Closing stream and exiting
+		log.Printf("Stream auto-close conditions met - exiting")
+		ss.Close()
```

Comment:
Just to confirm: this indeed flushes the file write buffer, right?

Reply:
Yes, inside Close there is a CloseFileWriter method which flushes it.